tactics/tactics.ml

```diff
@@ -1606,7 +1606,9 @@ let descend_in_conjunctions avoid tac (err, info) c =
                   (assert_before_gen false (NamingAvoid avoid) pt)
                   (fun _ -> Proofview.tclZERO ~info err))
                 [Proofview.tclORELSE
-                   (refiner ~check:true EConstr.Unsafe.(to_constr p))
+                  (Proofview.Goal.enter begin fun gl ->
```

Comment:
Why do we need to enter again here?

Reply:
This is not the same goal, it runs after the `assert` tactic.

Reply:
Note that it could probably be replaced by a more atomic call to a variant of `pose` but that would require more re-engineering at the risk of introducing incompatibilities. The main motivation for this PR was to strictly reduce the number of calls to the old refiner.